### PR TITLE
34 create flutter objects class

### LINKED ISSUE #34: Create Flutter Objects Class
**Short description:**
Create Flutter objects to store all data from the DB to use it in our app

**Describe the solution you'd like**

Create Flutter objects which get all data from JSON with an API call

Objects : 

- [x] 1. Error

> 1. Error message

- [x]  2. User

>1. Username
>2. E-mail
>3. Token auth
>4. is Admin
>5. Created At.

- [x] 3. Service

>1. Name
>2. Description
>3. Created At.

- [x] 4. Action

>1. Name
>2. Description
>3. Created At.
>4. Is Enable.

- [x] 5. Reaction

>1. Name
>2. Description
>3. Created At.
>4. Is Enable.

Add tests for classes : 

- [x] User
- [x] Services
- [x] Action
- [x] Reaction
- [x] Error

### PR DESCRIPTION
**Pull request by TomDUVAL-MAHE**
**Short description:**
Create Flutter objects to store all data from the DB to use it in our app

**Describe the solution you'd like**

Create Flutter objects which get all data from JSON with an API call

Objects : 

- [x] 1. Error

> 1. Error message

- [x]  2. User

>1. Username
>2. E-mail
>3. Token auth
>4. is Admin
>5. Created At.

- [x] 3. Service

>1. Name
>2. Description
>3. Created At.

- [x] 4. Action

>1. Name
>2. Description
>3. Created At.
>4. Is Enable.

- [x] 5. Reaction

>1. Name
>2. Description
>3. Created At.
>4. Is Enable.

Add tests for classes : 

- [x] User
- [x] Service
- [x] Action
- [x] Reaction
- [x] Error

### CODE REVIEW

**Review by ZQUEMA — DISMISSED** [review]

File: application/app_code/lib/flutter_objects/action_data.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/// This class is the action class.
+/// It contains all information about a action
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/action_data.dart]: An ?

File: application/app_code/lib/flutter_objects/action_data.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+class ActionData {
+  final String name;
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/action_data.dart]: Final ? See discord

File: application/app_code/lib/flutter_objects/action_data.dart [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    required this.createdAt,
+    required this.isEnable,
+  });
+
+  /// Convert a json map into the class
+  factory ActionData.fromJson(Map<String, dynamic> json) {
+    return ActionData(
+        name: json['action']['name'],
+        description: json['action']['description'],
+        createdAt: json["action"]["createdAt"],
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/action_data.dart]: Please use simple quote

File: application/app_code/lib/flutter_objects/error_data.dart [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+  /// Constructor of the ErrorData class
+  const ErrorData({
+    required this.errorMessage,
+  });
+
+  /// Convert a json map into the class
+  factory ErrorData.fromJson(Map<String, dynamic> json) {
+    return ErrorData(
+      errorMessage: json['errorMessage'],
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/error_data.dart]: An error does'nt have a field

File: application/app_code/lib/flutter_objects/reaction_data.dart [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    required this.createdAt,
+    required this.isEnable,
+  });
+
+  /// Convert a json map into the class
+  factory ReactionData.fromJson(Map<String, dynamic> json) {
+    return ReactionData(
+        name: json['reaction']['name'],
+        description: json['reaction']['description'],
+        createdAt: json["reaction"]["createdAt"],
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/reaction_data.dart]: Quote

File: application/app_code/lib/flutter_objects/service_data.dart [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    required this.description,
+    required this.createdAt,
+  });
+
+  /// Convert a json map into the class
+  factory ServiceData.fromJson(Map<String, dynamic> json) {
+    return ServiceData(
+      name: json['service']['name'],
+      description: json['service']['description'],
+      createdAt: json["service"]["createdAt"],
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/service_data.dart]: Quote

File: application/app_code/lib/flutter_objects/user_data.dart [hunk 7]
```diff
@@ -0,0 +1,28 @@
+/// This class is the UserData class.
+/// It contains all information about a UserData
+class UserData {
+  final String userName;
+  final String token;
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/user_data.dart]: Make it null possible with ?

File: application/app_code/lib/flutter_objects/user_data.dart [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  final bool isAdmin;
+  final String email;
+  final DateTime createdAt;
+
+  /// Constructor of the UserData class
+  const UserData({
+    required this.userName,
+    required this.email,
+    required this.token,
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/user_data.dart]: Not required

File: application/app_code/lib/flutter_objects/user_data.dart [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    required this.createdAt,
+  });
+
+  /// Convert a json map into the class
+  factory UserData.fromJson(Map<String, dynamic> json) {
+    return UserData(
+        userName: json['user']['username'],
+        email: json['user']['email'],
+        isAdmin: json['user']['isAdmin'],
+        createdAt: json["user"]["createdAt"],
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/user_data.dart]: Quote

File: application/app_code/test/testsFlutterObjects/action_data_tests.dart [hunk 10]
```diff
@@ -0,0 +1,68 @@
+import 'package:application/flutter_objects/action_data.dart';
+import 'package:application/main.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void templateFunction() {
```
Comment (LucasTesnier) [inline application/app_code/test/testsFlutterObjects/action_data_tests.dart]: Why did you put it in the template function ? 

Please create one function for each object.

By the way, try to respect the projet architecture in the tests directory

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
A lot of minor things.
By the way, your unit testing can't work !
You only need to have one main function calling all your unit function.
Go to discord for more informations

File: application/app_code/lib/flutter_objects/user_data.dart [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  /// Convert a json map into the class
+  factory UserData.fromJson(Map<String, dynamic> json) {
+    return UserData(
+        userName: json['user']['username'],
+        email: json['user']['email'],
+        isAdmin: json['user']['isAdmin'],
+        createdAt: json['user']['createdAt'],
+        token: json['token']);
```
Comment (LucasTesnier) [inline application/app_code/lib/flutter_objects/user_data.dart]: make it optional

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Just one minor things, will talk about this right after the meeting

**Review by LucasTesnier — APPROVED** [review]

**Review by ZQUEMA — APPROVED** [review]

**Pull request merged**